keps/sig-cluster-lifecycle/0032-create-a-k8s-io-component-repo.md

```diff
@@ -0,0 +1,245 @@
+---
+kep-number: 32
+title: Create a `k8s.io/component` repo
+status: implementable
+authors:
+  - "@luxas"
+  - "@sttts"
+owning-sig: sig-cluster-lifecycle
+participating-sigs:
+  - sig-api-machinery
+  - sig-cloud-provider
+reviewers:
+  - "@thockin"
+  - "@jbeda"
+  - "@bgrant0607"
+  - "@smarterclayton"
+  - "@liggitt"
+  - "@lavalamp"
+  - "@andrewsykim"
+  - "@cblecker"
+approvers:
+  - "@thockin"
+  - "@jbeda"
+  - "@bgrant0607"
+  - "@smarterclayton"
+editor:
+  name: "@luxas"
+creation-date: 2018-11-27
+last-updated: 2018-11-27
+---
+
+# Create a `k8s.io/component` repo
+
+**How we can consolidate the look and feel of core and non-core components with regards to ComponentConfiguration, flag handling, and common functionality with a new repository**
+
+## Table of Contents
+
+- [Create a `k8s.io/component` repo](#create-a--k8sio-component--repo)
+  - [Table of Contents](#table-of-contents)
+  - [Abstract](#abstract)
+    - [History and Motivation](#history-and-motivation)
+    - ["Component" definition](#-component--definition)
+    - [Goals](#goals)
+    - [Success metrics](#success-metrics)
+    - [Non-goals](#non-goals)
+    - [Related proposals / references](#related-proposals---references)
+  - [Proposal](#proposal)
+    - [Part 1: ComponentConfig](#part-1--componentconfig)
+      - [Standardized encoding/decoding](#standardized-encoding-decoding)
+      - [Testing helper methods](#testing-helper-methods)
+      - [Generate OpenAPI specifications](#generate-openapi-specifications)
+    - [Part 2: Command building / flag parsing](#part-2--command-building---flag-parsing)
+      - [Wrapper around cobra.Command](#wrapper-around-cobracommand)
+      - [Flag precedence over config file](#flag-precedence-over-config-file)
+      - [Standardized logging](#standardized-logging)
+    - [Part 3: HTTPS serving](#part-3--https-serving)
+      - [Common endpoints](#common-endpoints)
+      - [Standardized authentication / authorization](#standardized-authentication---authorization)
+    - [Part 4: Sample implementation in k8s.io/sample-component](#part-4--sample-implementation-in-k8sio-sample-component)
+    - [Code structure](#code-structure)
+    - [Timeframe and Implementation Order](#timeframe-and-implementation-order)
+    - [OWNERS file for new packages](#owners-file-for-new-packages)
+
+## Abstract
+
+The proposal is about preparing the Kubernetes core package structure in a way that all core component can share common code around
```

Comment:
you mention "core" here, but then don't repeat core elsewhere.  I think it's a mistake to continue the "same ideas but different implementation" or "just copy-and-paste-from-core" approach that's been taken in the past for external components.